<p>This rule detects lazy class. If your class doesn't work ,or it doesn't have any method. You will get the issue. It means you should Refactor your code to make sure that every class works.  </p>
<h2>Noncompliant Code Example</h2>
<pre>
TO DO 
</pre>
<h2>Compliant Solution</h2>
<pre>
TO DO 
</pre>
