<p>This rule detects LargeClass. If your class has over 150 lines. You will get the issue. It means you should Refactor your code to simplify the class.</p>
<h2>Noncompliant Code Example</h2>
<pre>
TO DO 
</pre>
<h2>Compliant Solution</h2>
<pre>
TO DO 
</pre>
