<p>This rule detects refused bequest. If child class inherits parent class child class doesn't use most methods and variable in parent class. You will get the issue. It means you should make sure that is it really have to use Inheritance relationship. </p>
<h2>Noncompliant Code Example</h2>
<pre>
TO DO 
</pre>
<h2>Compliant Solution</h2>
<pre>
TO DO 
</pre>
